protocols/gossipsub/src/config.rs

```diff
@@ -231,8 +231,19 @@ impl GossipsubConfigBuilder {
 
 impl std::fmt::Debug for GossipsubConfig {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        struct StringOrBytes<'a>(&'a [u8]);
+
+        impl<'a> std::fmt::Debug for StringOrBytes<'a> {
+            fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                if let Ok(text) = std::str::from_utf8(self.0) {
+                    write!(f, "\"{}\"", text)
+                } else {
+                    write!(f, "{}", hex_fmt::HexFmt(self.0))
+                }
+            }
+        }
         let mut builder = f.debug_struct("GossipsubConfig");
-        let _ = builder.field("protocol_id", &self.protocol_id);
+        let _ = builder.field("protocol_id", &StringOrBytes(&self.protocol_id));
```

Comment:
It feels more simple to me to just inline that code.

(I typed that blindly, might not compile)

```suggestion
        if let Ok(text) = std::str::from_utf8(self.protocol_id.0) {
            let _ = builder.field("protocol_id", &text));
        } else {
            let _ = builder.field("protocol_id", hex_fmt::HexFmt(self.protocol_id.0));
        }
```

Reply:
You are right. I had that thing laying around and copied it in, but here it is only used once.